libnopegl/src/pgcraft.c

```diff
@@ -1428,11 +1427,11 @@ struct pgcraft *ngli_pgcraft_create(struct ngl_ctx *ctx)
     ngli_darray_init(&s->filtered_pipeline_info.desc.vertex_buffers, sizeof(struct vertex_buffer_layout), 0);
 
     ngli_darray_init(&s->pipeline_info.data.textures,   sizeof(struct texture *), 0);
-    ngli_darray_init(&s->pipeline_info.data.buffers,    sizeof(struct buffer *),  0);
+    ngli_darray_init(&s->pipeline_info.data.buffers,    sizeof(struct buffer_binding),  0);
     ngli_darray_init(&s->pipeline_info.data.vertex_buffers, sizeof(struct buffer *),  0);
 
     ngli_darray_init(&s->filtered_pipeline_info.data.textures,   sizeof(struct texture *), 0);
-    ngli_darray_init(&s->filtered_pipeline_info.data.buffers,    sizeof(struct buffer *),  0);
+    ngli_darray_init(&s->filtered_pipeline_info.data.buffers,    sizeof(struct buffer_binding),  0);
     ngli_darray_init(&s->filtered_pipeline_info.data.vertex_buffers, sizeof(struct buffer *),  0);
```

Comment:
it probably makes sense to rename `buffers`, because we now have `buffers` and `vertex_buffers` with different types

Reply:
I would prefer to keep its name short. Also it aligns with the pipeline_compat.resources field name.